bgpd/bgp_main.c

```diff
@@ -407,6 +415,9 @@ int main(int argc, char **argv)
 	/* BGP related initialization.  */
 	bgp_init();
 
+	zclient->instance = instance;
```

Comment:
This will cause problems, We need to tell *zebra* what instance we are using  if we are going to set the instance.  As such bgp_init() should be changed to pass the instance so that bgp_zebra_init can be passed in the correct instance.